src/knx/usb_tunnel_interface.cpp

```diff
@@ -0,0 +1,543 @@
+		{
+			println("Error: Wrong sequence number!");
+			delete data;
+			continue;
+		}
+
+		// first RX buffer from queue should contain the first part of the transfer protocol packet
+		if ((expSeqNum == 1) && ((packetType & PACKET_TYPE_START) != PACKET_TYPE_START))
+		{
+			println("Error: Sequence number 1 does not contain a START packet!");
+			delete data;
+			continue;
+		}
+
+		// Make sure we only have one START packet
+		if ((expSeqNum != 1) && ((packetType & PACKET_TYPE_START) == PACKET_TYPE_START))
+		{
+			println("Error: Sequence number (!=1) contains a START packet!");
+			delete data;
+			continue;
+		}
+
+		// Make sure other packets are marked correctly as PARTIAL packet
+		if ((expSeqNum != 1) && ((packetType & PACKET_TYPE_PARTIAL) != PACKET_TYPE_PARTIAL))
+		{
+			println("Error: Sequence number (!=1) must be a PARTIAL packet!");
+			delete data;
+			continue;
+		}
+
+		// Not really necessary, but we reset the offset here to zero
+		if ((packetType & PACKET_TYPE_START) == PACKET_TYPE_START)
+		{
+			offset = 0;
+		}
+
+		// Copy KNX HID Report Body to final buffer for concatenating
+		memcpy(&tpPacket[offset], &data[3], packetLength);
+		// Remove the source HID report buffer
+		delete data;
+		// Move offset
+		offset += packetLength;
+
+		// If we reached the end of the transport protocol packet, leave the loop
+		if ((packetType & PACKET_TYPE_END) == PACKET_TYPE_END)
+		{
+			success = true;
+			break;
+		}
+	}
+
+	// Make sure that we really saw the end of the transport protocol packet
+	if (success)
+	{
+		handleTransferProtocolPacket(tpPacket, offset);
+	}
+	else
+	{
+		println("Error: Did not find END packet!");
+	}
+}
+
+void UsbTunnelInterface::handleBusAccessServerProtocol(ServiceIdType servId, const uint8_t* requestData, uint16_t packetLength)
+{
+	uint8_t respData[3]; // max. 3 bytes are required for a response
+
+	switch (servId)
+	{
+		case DeviceFeatureGet: // Device Feature Get
+		{
+			FeatureIdType featureId = (FeatureIdType)requestData[0];
+			respData[0] = (uint8_t) featureId; // first byte in repsonse is the featureId itself again
+
+			switch (featureId)
+			{
+				case SupportedEmiType: // Supported EMI types
+					Serial1.println("Device Feature Get: Supported EMI types");
+					respData[1] = 0x00; // USB KNX Transfer Protocol Body: Feature Data
+					respData[2] = 0x04; // USB KNX Transfer Protocol Body: Feature Data -> only cEMI supported
+					sendKnxHidReport(BusAccessServer, DeviceFeatureResponse, respData, 3);
+					break;
+				case HostDeviceDescriptorType0: // Host Device Descriptor Type 0
+					Serial1.println("Device Feature Get: Host Device Descriptor Type 0");
+					pushWord(_maskVersion, &respData[1]); // USB KNX Transfer Protocol Body: Feature Data -> Mask version
+					sendKnxHidReport(BusAccessServer, DeviceFeatureResponse, respData, 3);
+					break;
+				case BusConnectionStatus: // Bus connection status
+					Serial1.println("Device Feature Get: Bus connection status");
```

Comment:
The Serial1.println should probably be only println().

Reply:
yes. fixed.

Reply:
Ok. I will change it after I finish debugging the new memory stuff.